src/Button/Button.jsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+
+require('./Button.less');
+
+const VALID_TYPES = ["primary", "secondary", "destructive"];
+const VALID_SIZES = ["large", "regular", "small", "regular"];
+
+export class Button extends React.Component {
+  constructor(props) {
+    super();
+    if (props.type && VALID_TYPES.indexOf(props.type) === -1) {
```

Comment:
We could get most of this validation for free/cheap if we declare propTypes like @osdiab mentioned. For example:
```javascript
Button.propTypes = {
  type: React.PropTypes.oneOf(['primary', 'secondary', 'destructive']).isRequired,
  size: React.PropTypes.oneOf(['large', 'regular', 'small']),
  value: React.PropTypes.string.isRequired,
  disabled: React.PropTypes.bool
}
```